nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-properties-loader/src/main/java/org/apache/nifi/properties/NiFiPropertiesLoader.java

```diff
@@ -211,6 +213,25 @@ public NiFiProperties get() {
         return instance;
     }
 
+    private void checkForDuplicates(File file) {
+        if (file == null || !file.exists() || !file.canRead()) {
+            throw new IllegalArgumentException("NiFi properties file missing or unreadable");
+        }
+
+        Props properties = new Props();
+        try (final InputStream inputStream = new BufferedInputStream(new FileInputStream(file))) {
+            properties.load(inputStream);
+        } catch (final Exception e) {
+            throw new RuntimeException(String.format("Loading Application Properties [%s] failed", file), e);
+        }
+
+        Set<String> duplicateKeys = properties.duplicateKeySet();
+        if (!duplicateKeys.isEmpty()) {
```

Comment:
With the change to use an extension of `java.util.Properties`, recommend relocating this check into `loadProtectProperties()`. That would avoid the need to load the file twice, and also remove the duplicate file existence checking.

Reply:
Updated the location of the check to avoid duplicate loading/parsing of properties file.